platform/src/main/java/org/stellar/anchor/platform/payment/observer/stellar/StellarPaymentObserver.java

```diff
@@ -40,35 +47,69 @@ public class StellarPaymentObserver implements HealthCheckable {
   /** The minimum number of results the Stellar Blockchain can return. */
   private static final int MIN_RESULTS = 1;
 
+  // If the observer had been silent for longer than MAX_SILENT_DURATION, it will be marked
+  // unhealthy.
+  private static final long MAX_SILENT_TIMEOUT = 90;
+
+  // The time interval between silence checks
+  private static final long SILENCE_CHECK_INTERVAL = 5;
+
   final Server server;
-  final Set<PaymentListener> observers;
+  final Set<PaymentListener> paymentListeners;
   final StellarPaymentStreamerCursorStore paymentStreamerCursorStore;
   final Map<SSEStream<OperationResponse>, String> mapStreamToAccount = new HashMap<>();
   final PaymentObservingAccountsManager paymentObservingAccountsManager;
   SSEStream<OperationResponse> stream;
 
-  private final ExponentialBackoffTimer exponentialBackoffTimer = new ExponentialBackoffTimer();
+  final ExponentialBackoffTimer exponentialBackoffTimer = new ExponentialBackoffTimer();
+  boolean healthy = true;
+  Instant lastActivityTime;
+
+  ScheduledExecutorService watcher = Executors.newSingleThreadScheduledExecutor();
 
   StellarPaymentObserver(
       String horizonServer,
-      Set<PaymentListener> observers,
+      Set<PaymentListener> paymentListeners,
       PaymentObservingAccountsManager paymentObservingAccountsManager,
       StellarPaymentStreamerCursorStore paymentStreamerCursorStore) {
     this.server = new Server(horizonServer);
-    this.observers = observers;
+    this.paymentListeners = paymentListeners;
     this.paymentObservingAccountsManager = paymentObservingAccountsManager;
     this.paymentStreamerCursorStore = paymentStreamerCursorStore;
   }
 
   /** Start watching the accounts. */
   public void start() {
-    this.stream = watch();
+    this.stream = startSSEStream();
+    this.healthy = true;
+    infoF("Starting the observer watcher");
+    watcher.scheduleAtFixedRate(
+        this::watchTransactionTimeout,
+        10,
+        SILENCE_CHECK_INTERVAL,
+        TimeUnit.SECONDS); // TODO: The period should be made configurable in version 2.x
+  }
+
+  private void watchTransactionTimeout() {
+    Instant now = Instant.now();
+    if (lastActivityTime != null) {
+      Duration silenceDuration = Duration.between(lastActivityTime, now);
+      debugF("The observer had been silent for {} seconds", silenceDuration.getSeconds());
+      if (silenceDuration.getSeconds() > MAX_SILENT_TIMEOUT) {
+        infoF(
+            "The observer had been silent for {} seconds. Marking it unhealthy",
+            silenceDuration.getSeconds());
+        this.healthy = false;
+      }
+    }
```

Comment:
Have you explored the `onClosed` listener? That could give you a simpler and more reliable way to identify that a connection has been closed, so you can restart it.